math/mathcore/inc/Fit/FitUtil.h

```diff
@@ -259,1212 +207,174 @@ namespace FitUtil {
-
-         return FitUtil::EvaluateChi2(func, data, p, nPoints, executionPolicy, nChunks);
-      }
-
-      static double EvalLogL(const IModelFunctionTempl<double> &func, const UnBinData &data, const double *p,
-                             int iWeight, bool extended, unsigned int &nPoints,
-                             ::ROOT::Fit::ExecutionPolicy executionPolicy, unsigned nChunks = 0)
-      {
-         return FitUtil::EvaluateLogL(func, data, p, iWeight, extended, nPoints, executionPolicy, nChunks);
-      }
-
-      static double EvalPoissonLogL(const IModelFunctionTempl<double> &func, const BinData &data, const double *p,
-                                    int iWeight, bool extended, unsigned int &nPoints,
-                                    ::ROOT::Fit::ExecutionPolicy executionPolicy, unsigned nChunks = 0)
-      {
-         return FitUtil::EvaluatePoissonLogL(func, data, p, iWeight, extended, nPoints, executionPolicy, nChunks);
-      }
-
-      static double EvalChi2Effective(const IModelFunctionTempl<double> &func, const BinData & data, const double * p, unsigned int &nPoints)
-      {
-         return FitUtil::EvaluateChi2Effective(func, data, p, nPoints);
-      }
-      static void EvalChi2Gradient(const IModelFunctionTempl<double> &func, const BinData &data, const double *p,
-                                   double *g, unsigned int &nPoints,
-                                   ::ROOT::Fit::ExecutionPolicy executionPolicy = ::ROOT::Fit::ExecutionPolicy::kSerial,
-                                   unsigned nChunks = 0)
-      {
-         FitUtil::EvaluateChi2Gradient(func, data, p, g, nPoints, executionPolicy, nChunks);
-      }
-      static double EvalChi2Residual(const IModelFunctionTempl<double> &func, const BinData & data, const double * p, unsigned int i, double *g = 0)
-      {
-         return FitUtil::EvaluateChi2Residual(func, data, p, i, g);
-      }
+   // function to avoid infinities or nan
+   inline double CorrectValue(double rval)
+   {
+      // avoid infinities or nan in  rval
+      if (rval > -std::numeric_limits<double>::max() && rval < std::numeric_limits<double>::max())
+         return rval;
+      else if (rval < 0)
+         // case -inf
+         return -std::numeric_limits<double>::max();
+      else
+         // case + inf or nan
+         return +std::numeric_limits<double>::max();
+   }
 
-      /// evaluate the pdf (Poisson) contribution to the logl (return actually log of pdf)
-      /// and its gradient
-      static double EvalPoissonBinPdf(const IModelFunctionTempl<double> &func, const BinData & data, const double *p, unsigned int i, double *g ) {
-         return FitUtil::EvaluatePoissonBinPdf(func, data, p, i, g);
+   // Check if the value is a finite number. The argument rval is updated if it is infinite or NaN,
+   // setting it to the maximum finite value (preserving the sign).
+   inline bool CheckInfNaNValue(double &rval)
```

Comment:
Please use `std::isfinite` instead of this. https://en.cppreference.com/w/cpp/numeric/math/isfinite